packages/graphback-datasync/src/providers/DeltaProvider.ts

```diff
@@ -0,0 +1,203 @@
+import { applyIndexes } from "@graphback/runtime-mongo";
+import { GraphQLObjectType } from 'graphql';
+import { Db, ObjectId, FilterQuery } from "mongodb"
+import { getTableName, GraphbackOperationType } from '@graphback/core';
+
+function getDeltaTableName(baseType: GraphQLObjectType) {
+  return `Graphback-${getTableName(baseType)}-Delta`
+}
+
+type DeltaOps = GraphbackOperationType.CREATE | GraphbackOperationType.UPDATE | GraphbackOperationType.DELETE;
+
+export interface ConflictingFieldsReport {
+  _id: ObjectId,
+  conflicts: {
+    [fieldName: string]: any
+  },
+  version: number
+}
+
+interface DeltaDocument {
+  docId: ObjectId,
+  op: DeltaOps,
+  timestamp: Date,
+  version: number,
+  sets: {
+    [fieldName: string]: any
+  }
+}
+
+
+/**
+ * Provides an interface for interacting with deltas and stuff
```

Comment:
Would this be used for data read? Are delta served from this?

Reply:
Delta being served from this is very bad from multiple standpoints, I had written this code before we had the discussion on #1744 , this is definitely going away

Reply:
Cool. Sorry for commenting in early phase.